src/backend/app/auth/auth_routes.py

```diff
@@ -218,3 +218,15 @@ async def my_data(
         user_data(dict): The dict of user data.
     """
     return await get_or_create_user(db, user_data)
+
+
+@router.get("/introspect")
+async def check_login(
+    db: Session = Depends(database.get_db),
+    user_data: AuthUser = Depends(login_required),
+):
+    """Verifies the validity of login cookies.
+
+    Returns True if authenticated, False otherwise.
+    """
```

Comment:
Good stuff 😄

Thinking about it, the endpoint may as well return the actual user data json, as the frontend could then update the profile picture if it changed since last login 👍

Reply:
okay